R-Python-Testing/write_python_read_r.py

```diff
@@ -0,0 +1,75 @@
+import shutil
+import subprocess
+
+import pyarrow as pa
+
+import tiledbsoma as soma
+
+# Remove artifacts from previous runs, if they exist
+uri = "test-dataframe.auto.soma"
+shutil.rmtree(uri)
+
+asch = pa.schema(
+    [
+        ("foo", pa.int32()),
+        ("bar", pa.float64()),
+        ("baz", pa.large_string()),
+        ("quux", pa.bool_()),
+    ]
+)
+
+soma.DataFrame.create(uri, schema=asch, index_column_names=["foo"]).close()
+
+pydict = {}
+pydict["soma_joinid"] = [0, 1, 2, 3, 4]
+pydict["foo"] = [10, 20, 30, 40, 50]
+pydict["bar"] = [4.1, 5.2, 6.3, 7.4, 8.5]
+pydict["baz"] = ["apple", "ball", "cat", "dog", "egg"]
+pydict["quux"] = [True, False, False, True, False]
+rb = pa.Table.from_pydict(pydict)
+with soma.DataFrame.open(uri, "w") as sdf:
+    sdf.write(rb)
+
+df = rb.to_pandas()  # Alternatively, we could re-open the dataframe
+
+# Prepares an R script with the dependencies and loads the data.frame in `df`.
+# Future assertions can be done by appending to this script
+R_script_base = f"""
+require("tiledbsoma")
+require("testthat")
+platform_config <- NULL
+tiledbsoma_ctx <- NULL
+soma_df <- SOMADataFrame$new("{uri}", platform_config, tiledbsoma_ctx)
+table = soma_df$read()
+df = as.data.frame(table)
+"""
+
+
+def r_assert(code: str):
+    R_script = R_script_base + code
+    with open("test-dataframe-read.R", "w") as f:
+        f.write(R_script)
+    subprocess.run(["Rscript", "test-dataframe-read.R"])
```

Comment:
I don't understand where the assertion failure is caught.  If I read this right, the Rscript process will return a non-zero exit code, but it will be ignored by `subprocess.run` because it doesn't have the `check` param set to True, AND/OR there is no check for non-zero exit code.

I was imagining that the `r_assert` would raise an exception (like `assert`) when the R script fails.

So maybe this line should be:

> assert subprocess.run(["Rscript", "test-dataframe-read.R"]).returncode == 0

or something like that?